src/CIManager.ts

```diff
@@ -132,28 +145,83 @@ export default class CIManager {
         }
         return await tokenCache.token;
     }
-    async launchBuildJob(req: {
-        owner: string,
-        repositoryName: string,
-        repositoryURL: string
-        labels: string[]
-    }) {
-        console.log(req);
+    async launchBuildJob(req: LauncherState) {
         const orgInstallationOctokit = this._installations.find(v => v.orgName == req.owner)?.octokit;
         if (!orgInstallationOctokit) {
             console.log("No app installation found for org " + req.owner);
             return;
         }
-        const launcherToken = await jwt.sign({ "req": req , "createdAt": Date.now()}, this._jwtKey);
-        this._redis.sAdd(req.repositoryURL+req.labels.join(","), launcherToken);
-
-       
-        console.log(launcherToken)
+        const launcherToken = await jwt.sign(req, this._jwtKey);
+        this._redis.zAdd(req.repositoryURL + req.labels.join(","), { score: Date.now(), value: req.workflowId.toString() });
         this.launchWorker({
-            gitHubURL: req.repositoryURL, launcherToken: launcherToken,
-            runnerLabels: req.labels.join(',')
+            gitHubURL: req.repositoryURL,
+            launcherToken: launcherToken,
+            runnerLabels: req.labels.join(','),
+            tokenURL: process.env.RUNNER_URL || '',
         })
     }
+    async removeBuildJob(req: LauncherState) {
+        return await this._redis.zRem(req.repositoryURL + req.labels.join(","), req.workflowId.toString());
+    }
+    async updateInflux() {
+        const writeApi = this._influx.getWriteApi(org, bucket);
+        writeApi.useDefaultTags({ host: 'host1' });
+
+        const keys = await this.getAllKeys();
+
+        let numJobs = 0;
+        let oldestJob = (await this._redis.zRangeWithScores(keys.pop() || "", 0, 0))[0];
+
+        keys.forEach(async key => {
```

Comment:
This is unlikely to be correct, because the promise in each of the loop iterations (returned by `async key =>` will be ignored). To get the intended effect, try:
```
for(const key of keys){
    const jobs = (await ...)
}
```